website/docs: add procedural docs for RAC (#9006)
goauthentik/authentik: website/docs/providers/rac/how-to-rac.md

---
title: Create a Remote Access Control (RAC) provider
---

:::info
This feature is in technical preview, so please report any bugs on [GitHub](https://github.com/goauthentik/authentik/issues).
:::

The RAC provider is a highly flexible feature for accessing remote machines. This document provides instructions for the basic creation and configuration of a RAC provider within a defined scenario.

> Fow more information about using a RAC provider, see the [Overview](./index.md) documentation. You can also view our [video on YouTube](https://www.youtube.com/watch?v=9wahIBRV6Ts) for setting up a RAC.

Reviewer: we could probably embed the video here like on the install page?

Reviewer: ```suggestion
For more information about using a RAC provider, see the [Overview](./index.md) documentation. You can also view our [video on YouTube](https://www.youtube.com/watch?v=9wahIBRV6Ts) for setting up RAC.
```

The first step is to create the RAC app and provider.

1. Log in as an admin to authentik, and go to the Admin interface.

2. In the Admin interface, navigate to **Applications -> Applications**.

3. Click **Create with Wizard**. Follow the [instructions](../../applications/manage_apps.md#instructions) to create your RAC application and provider.

### Step 2. Create RAC property mapping

Next, you need to add a property mapping for each of the remote machines you want to access. Property mappings allow you to pass information to external applications, and with RAC they are used to pass the host name, IP address, and access credentials for the remote machines.

1. In the Admin interface, navigate to **Customization -> Property Mappings**.

2. On the **Property Mappings** page, click **Create**.

3. On the **New property mapping** modal, set the following:

    - **Select Type**: RAC Property Mappings
    - **Create RAC Property Mapping**:
        - **Name**s: define a name for the property mapping, perhaps include the type of connection (RDP, SSH, VNC)
        - **General settings**:
            - **Username**: the username for the remote machine

>             - **Password**: the password for remote machines

Reviewer 2: Typo? One of these is plural, "... for the remote machine," the other singular: "...for remote machines." I find that very confusing without explanation or context.

        - **RDP settings**:

>             - **Ignore server certificate: select **Enabled\*\* (This setting is required for TRAC to work)

Reviewer: ```suggestion
            - **Ignore server certificate: select **Enabled\*\* (Depending on the setup of your RDP Server, it might be required to enable this setting.)
```

            - **Enable wallpaper**: optional

>             - **Enable font smooting**: optional

Reviewer 2: Typo: "font smooting" -> "smoothing."  I don't think you mean "a small hole in a wall that allows sheep to pass through."

            - **Enable full window dragging**: optional
        - Advanced settings:
            - **Expressions**: optional, using Python you can define custom [expressions](../../property-mappings/expression.mdx).

4. Click **Finish** to save your settings and close the modal.

### Step 3. Create Endpoints for the Provider

Finally, you need to create an endpoint for each remote machine. Endpoints are defined within providers; connections between the remote machine and authentik are enabled through communication between the provider's endpoint and the remote machine.

1. In the Admin interface navigate to **Applications -> Providers**.

2. Select the RAC provider you created in Step 1 above.

3. On the Provider page, under **Endpoints**, click **Create**.

4. On the **Create Endpoint** modal, provide the following settings:

>     - **Name**s: define a name for the endpoint, perhaps include the type of connection (RDP, SSH, VNC)

Reviewer 2: Is that `s` after `Name` correct?

    - **Protocol**: select the appropriate protocol
    - **Host**: the host name or IP address of the system you are connecting to.

>       **Maximum concurrent connections**: select a value or use `-1` to disable the limitation.

Reviewer: ```suggestion
    - **Maximum concurrent connections**: select a value or use `-1` to disable the limitation.
```

>       ** Property mapping**: select either the property mapping that you created in Step 2, or use one of the default settings.

Reviewer 2: Loose whitespace before `Property mapping`.

>       **Property mapping**: select either the property mapping that you created in Step 2, or use one of the default settings.

Reviewer: ```suggestion
    - **Property mapping**: select either the property mapping that you created in Step 2, or use one of the default settings.
```

>       **Advance settings**: optional

Reviewer: ```suggestion
    - **Advance settings**: optional
```

5. Click **Create** to save your settings and close the modal.

### Access the remote machine

> To verify your configuration and access the remote machine, go to the **User interface** of you authentik instance. On the **My applications** page click the **Remote Access** application to display the defined endpoint(s).

Reviewer 2: Typo: "... of you..." , should be "... of your authentik instance."

> Click the endpoint for the remote machine that you want to access. authentik connects you to a secure shell on the remote machine, in your web browser.

Reviewer 2: Question: Do we lowercase authentik even when it starts a sentence? (We need a brand guide!)

Author: Yep, always lowercase... I try not to start sentences with our product name, but...  and Yeah, we have a baby Style Guide, do need to make an explicit Brnadoing section within it.

> Click the endpoint for the remote machine that you want to access. authentik connects you to a secure shell on the remote machine, in your web browser.

Reviewer: When only a single endpoint is configured/available to the user due to permissions, there is no additional dialog where the endpoint is chosen

Author: Ah, @BeryJu the only way I have seen it done is with Chris' video, and he had multiple endpoints. so what happens if there is only a single endpoint...? Does only clicking the RAC app on the library page take you straight to that one remote machine?

Author: Take a look at my rewording and let me know if that works. Thanks!
